src/components/ComparisonList.jsx

```diff
@@ -3,27 +3,37 @@ import { connect } from '../store/Connect';
 import { DeleteIcon } from './Icons.jsx';
 import '../../sass/components/_comparison-list.scss';
 
+const ComparisonListItem = ({
+    item,
+    AnimatedPrice,
+    onDeleteItem,
+}) => (
+    <div className="cq-comparison-item">
+        <span className="left">
+            <span className="cq-comparison-swatch" style={{ backgroundColor: item.color }} />
+            <span className="cq-comparison-label">{item.symbolObject.name}</span>
+        </span>
+        <span className="right">
+            <AnimatedPrice />
+            <DeleteIcon className="ciq-close" onClick={() => onDeleteItem(item.symbolObject)} />
+        </span>
+    </div>
+);
+
 const ComparisonList = ({
     comparisonSymbols,
     animatedPrices,
     onDeleteItem,
 }) => (
     <div className="cq-comparison">
-        {comparisonSymbols.map((item, i) => {
-            const AnimatedPrice = animatedPrices[i];
-            if (!AnimatedPrice) { return null; }
```

Comment:
@MahboobehMohammadi it is not removed. I separated it out as a `<ComparisonListItem/>` component

Reply:
@brucebinary I know. i meant this line ` if (!AnimatedPrice) { return null; }`.

Reply:
@MahboobehMohammadi owh. Hm... I've been adding and removing comparisons around. `AnimatedPrice` always has a value, since it is synced with the size of comparisons

I just add a minor refactor in a new commit to only update the animatedPrices array when length changes.